src/test/java/javaslang/collection/AbstractMapTest.java

```diff
@@ -99,6 +99,11 @@
         return IntMap.of(emptyMap());
     }
 
+    @Override
+    protected boolean emptyShouldBeSingleton() {
+        return false;
```

Comment:
No, the tests defined in `AbstractTraversableTest` would not pass with subclasses of `Map`, because `AbstractMapTest::empty()` returns `IntMap.of(emptyMap())`. The `Map` hierarchy has its own set of tests using `emptyMap()` instead!